s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/FileReader.java

```diff
@@ -52,58 +47,26 @@ public FileReader(final S3SourceConfig s3SourceConfig, final String bucketName,
 
     @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     Iterator<S3ObjectSummary> fetchObjectSummaries(final AmazonS3 s3Client) {
-        return new Iterator<>() {
-            private String continuationToken = null; // NOPMD
-            private List<S3ObjectSummary> currentBatch = new ArrayList<>();
-            private int currentIndex = 0; // NOPMD
-            private boolean isTruncated = true;
-
-            @Override
-            public boolean hasNext() {
-                // If there are unprocessed objects in the current batch, we return true
-                if (currentIndex < currentBatch.size()) {
-                    return true;
-                }
-
-                if (isTruncated) {
-                    fetchNextBatch();
-                    return !currentBatch.isEmpty();
-                }
-
-                return false;
-            }
-
-            @Override
-            public S3ObjectSummary next() {
-                if (!hasNext()) {
-                    return null;
-                }
-
-                return currentBatch.get(currentIndex++);
-            }
-
-            private void fetchNextBatch() {
-                currentBatch.clear();
-                currentIndex = 0;
-
-                final ListObjectsV2Request request = new ListObjectsV2Request().withBucketName(bucketName)
-                        .withMaxKeys(s3SourceConfig.getInt(FETCH_PAGE_SIZE) * PAGE_SIZE_FACTOR)
-                        .withContinuationToken(continuationToken);
-
-                final ListObjectsV2Result objectListing = s3Client.listObjectsV2(request);
-                currentBatch = objectListing.getObjectSummaries()
+        final ListObjectsV2Request request = new ListObjectsV2Request().withBucketName(bucketName)
+                .withMaxKeys(s3SourceConfig.getInt(FETCH_PAGE_SIZE) * PAGE_SIZE_FACTOR);
```

Comment:
This is where I thought we could make an improvement, by only querying after the last known processed file from S3.
We could have a config item for the first time, but after that we could track the last processed file so that we can query the next time after that file reducing the number of queries we need to make to S3 and starting faster.
At the moment we are streaming every file in the S3 bucket every time we callfetchObjectSummaries()

Reply:
This is a known issue and is a little out of scope of this PR. Will add `startAfter` property as discussed in next PRs.